packages/rule-no-broken-links/src/rule.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * @fileoverview This rule checks every link and image tag in the page and report if it is broken
+ * Checks for 404, 410, 500 or 503 status
+ */
+
+import { Category } from 'sonarwhal/dist/src/lib/enums/category';
+import { RuleContext } from 'sonarwhal/dist/src/lib/rule-context';
+import {
+    IRule,
+    ElementFound,
+    RuleMetadata,
+    IAsyncHTMLElement
+} from 'sonarwhal/dist/src/lib/types';
+import { debug as d } from 'sonarwhal/dist/src/lib/utils/debug';
+import { RuleScope } from 'sonarwhal/dist/src/lib/enums/rulescope';
+import { Requester } from 'sonarwhal/dist/src/lib/connectors/utils/requester';
+import { URL } from 'url';
+import { NetworkData } from 'sonarwhal/dist/src/lib/types';
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class NoBrokenLinksRule implements IRule {
+    public static readonly meta: RuleMetadata = {
+        docs: {
+            category: Category.accessibility,
+            description: `Rule to flag broken links in the page`
+        },
+        id: 'no-broken-links',
+        schema: [],
+        scope: RuleScope.site
+    };
+
+    public constructor(context: RuleContext) {
+        const requester = new Requester();
+        const brokenStatusCodes = [404, 410, 500, 503];
+
+        const handleRejection = async (error: any, absoluteUrl: string, linkElement: IAsyncHTMLElement) => {
+            debug(`Error accessing {$absoluteUrl}. ${JSON.stringify(error)}`);
+
+            await context.report(
+                absoluteUrl,
+                linkElement,
+                `Broken link found (404 response)`
+            );
+        };
+
+        const handleSuccess = async (networkData: NetworkData, absoluteUrl: string, linkElement: IAsyncHTMLElement) => {
+            const statusIndex = brokenStatusCodes.indexOf(
+                networkData.response.statusCode
+            );
+
+            if (statusIndex > -1) {
+                await context.report(
+                    absoluteUrl,
+                    linkElement,
+                    `Broken link found (${
+                        brokenStatusCodes[statusIndex]
+                    } response)`
+                );
+            }
+        };
+
+        // This method checks whether a url is broken and report if it is
+        const reportBrokenLink = async (hrefUrl: string, resource: any, linkElement: IAsyncHTMLElement) => {
+            const absoluteUrl = new URL(hrefUrl, resource).href;
+
+            await requester
```

Comment:
Change `await` for `return` and add return type to the function. You are already doing an `await` later when you call this function so no need to `await` here.